### ISSUE TITLE
Feature extraction with pre-trained TSM

### ISSUE THREAD

**Issue by olga-zats** [assembly-101/assembly101-action-recognition#1]
Hi!

I had a question concerning the TSM pre-extracted features. Since the TSM network requires 8 segments to perform shifting, which frames do you sample during the per-frame feature extraction? 
Thank you for your answer in advance and I look forward to hearing from you!

Best,
Olga

**Comment by dibschat** [comment]
Hi!

If you are asking about the per-frame TSM features for action anticipation/segmentation, we directly use the 2D backbone of the TSM (without any shifting operations) pretrained for action recognition to extract the features for each frame.
We also tried a (2D backbone + shifting) version where we densely sampled 8 consecutive frames in a sliding window manner and have the 2048-D segment feature correspond to the middle frame of the segment - but the performance improvement was extremely marginal. Hence, we sticked to the lightweight 2D backbone only.

**Comment by olga-zats** [comment]
Hi!

Thank you for the response, that's exactly what I meant :)

Best,
Olga

**Comment by aleflabo** [comment]
Hi! 

I'm trying to extract the features of a dataset after training the proposed TSM on it. Following the suggestion above, I should train the 2D backbone without shift, and this is what I'm trying:

`python main.py dataset RGB --arch resnet50 --num_segments 8 --gd 20 --lr 0.001 --lr_steps 20 40 --epochs 50 --batch-size 32 -j 16 --dropout 0.5 --consensus_type=avg --eval-freq=1 --npb --tune_from=pretrained_models/tsm_rgb_epic.ckpt --gpus 0 1
`
In this way the features in output of the base_model have a shape of (128, 2048). This is due to the number of segments, 128 = num_segments*batch_size. 

How should I handle this problem to save the correct 2048-D tensor per frame?

Kind regards,
Alessandro
